pkg/cluster/cluster.go

```diff
@@ -432,6 +432,12 @@ func (c *Cluster) compareStatefulSetWith(statefulSet *appsv1.StatefulSet) *compa
 		}
 	}
 
+	if !reflect.DeepEqual(c.Statefulset.Spec.Template.Spec.Affinity.NodeAffinity, statefulSet.Spec.Template.Spec.Affinity.NodeAffinity) {
+		needsReplace = true
+		needsRollUpdate = true
+		reasons = append(reasons, "new statefulset's NodeAffinity doesn't match the current one")
+	}
```

Comment:
This isn't already covered with a [previous](https://github.com/zalando/postgres-operator/blob/9dc5f2427d68150aa91dc5e9a26661d016551825/pkg/cluster/cluster.go#L375) check on affinity?

Reply:
You are right, it should be covered by it. I will remove those lines.